data/management/commands/run_schedules.py

```diff
@@ -113,11 +113,42 @@ def run_and_update_schedule(self, schedule, runInstance):
                 self.stdout.write('Script ran successfully for ' + schedule.script_name)
 
             # This makes sure the next runtime is calculated from the time the last one was run i.e current time
-            current_date_time = make_aware(datetime.now())
-            self.create_next_run_instance(schedule, current_date_time)
+            updated_runtime = self.update_next_runtime_to_previous_start_time(schedule, runInstance.start_at, make_aware(datetime.now()))
+            actual_runtime = self.get_most_recent_of_updated_runtime_or_expected_runtime(updated_runtime, runInstance, schedule)
+            self.create_next_run_instance(schedule, actual_runtime)
         except:
             self.update_run_instance(runInstance, 'e', 'An unexpected error occured while executing the script ... please contact the administrator')
-            self.create_next_run_instance(schedule, make_aware(datetime.now()))
+            updated_runtime = self.update_next_runtime_to_previous_start_time(schedule, runInstance.start_at, make_aware(datetime.now()))
+            actual_runtime = self.get_most_recent_of_updated_runtime_or_expected_runtime(updated_runtime, runInstance, schedule)
+            self.create_next_run_instance(schedule, actual_runtime)
+
+    def update_next_runtime_to_previous_start_time(self, schedule, instance_start_date, current_date_time):
+        next_runtime = self.calculate_next_runtime(
+            current_date_time,
+            schedule.interval,
+            schedule.interval_type
+        )
```

Comment:
Correct me if I'm wrong, but I think we only need this for this function... just needs to replace `current_date_time` on line 127 with `instance_start_date`

Reply:
You are right, using `instance_start_date` ensures we have the same start time for full cycle intervals (greater than 24hrs). Shorter intervals (mins, secs and hours) would have increasing times as these intervals are added to the previous start time.